src/Microsoft.AspNetCore.Routing/RequestDelegateRouteBuilderExtensions.cs

```diff
@@ -88,16 +145,52 @@ public static IRouteBuilder MapGet(this IRouteBuilder builder, string template,
         }
 
         /// <summary>
+        /// <para>
+        ///     This method is obsolete and will be removed in a future version. The recommended alternative is
+        ///     Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapMiddlewareGet.
+        /// </para>
+        /// <para>
+        ///     Adds a route to the <see cref="IRouteBuilder"/> that only matches HTTP GET requests for the given
+        ///     <paramref name="template"/>, and <paramref name="action"/>.
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The <see cref="IRouteBuilder"/>.</param>
+        /// <param name="template">The route template.</param>
+        /// <param name="action">The action to apply to the <see cref="IApplicationBuilder"/>.</param>
+        /// <returns>A reference to the <paramref name="builder"/> after this operation has completed.</returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.AspNetCore.Routing.RequestDelegateRouteBuilderExtensions.MapMiddlewareGet.")]
```

Comment:
Per discussion, we should probably get rid of these methods entirely to avoid the IntelliSense issue where the overloads are ambiguous at first.

Need to post a breaking-change announcements for this too.

cc @rynowak